lib/js_of_ocaml/js.ml

```diff
@@ -805,6 +805,8 @@ let export_js (field : js_string t) x =
     (Unsafe.pure_js_expr "jsoo_exports")
     field
     (if String.equal (Js.to_string (typeof (Obj.magic x))) "function"
+        (* function with arity/length equal to zero are already wrapped *)
+        && Unsafe.get (Obj.magic x) (Js.string "length") > 0
```

Comment:
I think this is the correct fix, and that we do not need to modify `caml_call_gen`.

Reply:
done